js/jquery.terminal.js

```diff
@@ -1357,6 +1358,8 @@
                     } else {
                         if (command !== '' && position > 0) {
                             self['delete'](-1);
+                            var str = $('.clipboard').val()
+                            $('.clipboard').val(str.substring(0, str.length - 1));
```

Comment:
Sorry I thought tat you removing the whole clipboard. 
I think that the clipboard deleteting should be moved to delete method and it should take into account cursor position or better fake_mobile_entry should be renamed, maybe to fix_clipboard and the `if (is_touch) {` should be removed from that function, this should fix the problem with input method.